### ISSUE TITLE
[Bug Report] Scan fails on every file in ParseDateStringAsFormat

### ISSUE THREAD

**Issue by Cygon** [stashapp/stash#694]
**Describe the bug**
I've set up Stash on my NAS with the executable as /opt/stash/stash-linux, its own system user called 'stash' and all generated + metadata stored in /var/cache/stash/{generated|metadata|temporary}.

My library is in several places, so I've created /opt/stash/media as the "stash" directory and put symlinks to the actual locations (readable by the stash user) inside.

It discovers my movies, but every single one fails with a problem in ParseDateStringAsFormat (and they leave behind orphaned thumbnail files, too...)

**To Reproduce**
1. Install Stash as documented (Gentoo Linux on a QNAP NAS system here)
2. Use standard paths from the FHS and XDG specs
3. Use one folder as your "stash" and put a symlink to another (readable) location with movies in there
4. Scan your library and view the log

```
/opt/stash/media/machinima/Some Video.mp4 doesn't exist.  Creating new item...
2020-07-28 18:05:31Debug   Creating screenshot for /opt/stash/media/machinima/Some Video.mp4
2020-07-28 18:05:31Debug   Creating thumbnail for /opt/stash/media/machinima/Some Video.mp4
2020-07-28 18:05:31Debug   Checksum calculated: 1445b3113cb4bf9969339cb852e29e85
2020-07-28 18:05:31Info    /opt/stash/media/machinima/Some Video.mp4 not found.  Calculating checksum...
2020-07-28 18:05:31Debug   sqlite date conversion error: ParseDateStringAsFormat failed: dateString <>, format <2006-01-02>
```

**Expected behavior**
I expect stash to be able to scan the movie files since it's finding them and also creating thumbnails already.

**Stash Version: (from Settings -> About):**

**Desktop (please complete the following information):**
 - Server OS: Gentoo Linux inside an LXC container on a QNAP NAS with kernel 4.14.24
 - Client OS: Gentoo Linux 5.4.38 w/KDE 5 Plasma
 - Browser: Firefox Quantum
 - Version: 68.9.0esr

**Additional context**
Maybe the symlink is to blame. Or I have an unexpected date format. Will try to check both as a follow-up.

**Comment by bnkai** [comment]
Are you sure the files are not added to your DB ? Did you try refreshing the scenes page?
The date error message is just a debug message it shouldn't affect anything.

**Comment by Cygon** [comment]
Yes, I hit Shift+Reload several times, the overview showed 0 Bytes, 0 Scenes/Movies/etc.

The ParseDateStringAsFormat message is the only indication I saw that something might have gone wrong.

When I added to actual directories of my media files (without symlinks) it immediately worked.

-----

I just retried the whole symlink setup. On the settings/tasks page it does show a progress bar, take a while and the log lists all the files behind the symlink, but the library stays empty.

So the bug (if even) is probably just that files behind symlinks fail to appear in the database. I can live with this, but it would be nice if symlinks could either work or be completely ignored :)

**Comment by bnkai** [comment]
Just finished retesting.
Under a media folder i created both softlinks to a  media file and a media directory (debian testing, latest stash dev version) . Files from both symlinks got scanned and added to my DB as expected. Are you sure there is no permissions issue. Did you create the softlinks with absolute paths?

**Comment by Cygon** [comment]
I must have messed up somewhere yesterday.

While attempting to pin down the issue, I first used a symlink to a directory in my LXC guest's file system... and it worked.

I then symlinked to a file in my movie library (which is mounted in a slightly exotic way, `/dev/mapper/cachedev3 on /srv/movies type ext4 (ro,relatime,lazytime,noacl,stripe=128,data=ordered,data_err=abort,jqfmt=vfsv0,usrjquota=aquota.user)` but this time, it worked, too.

Can't reproduce it anymore, even with the symlinks set up the same way as yesterday. No permissions changed, symlinks used absolute paths all along, checked my .bash_history for yesterday's commands without finding any mistakes, yet now it works... :clown_face:
